exasol_python_test_framework/udf/udf_debug.py

```
@@ -0,0 +1,169 @@
+        self.server_address = server_address
+        self.output = output
+
+        self.bind(self.server_address)
+
+        if self.server_address[1] == 0:
+            self.server_address = (self.server_address[0], self.socket.getsockname()[1])
+        self.listen(10)
+
+    def handle_accept(self):
+        conn, addr = self.accept()
+        LogHandler(conn, addr, self.output)
+
+    def handle_close(self):
+        self.close()
+
+
+class LogHandler(asynchat.async_chat):
+    def __init__(self, sock, address: Tuple, output: Queue):
+        asynchat.async_chat.__init__(self, sock=sock)
+        self.set_terminator(b"\n")
+        self.address = "%s:%d" % address
+        self.output = output
+        self.ibuffer = []
+
+    def collect_incoming_data(self, data):
+        self.ibuffer.append(data.decode('utf-8'))
+
+    def found_terminator(self):
+        try:
+            self.output.put_nowait("%s> %s\n" % (self.address, ''.join(self.ibuffer).rstrip()))
+        except Full as full_ex:
+            print(f"Queue is full for UDF debug:{full_ex}", file=sys.stderr)
+        self.ibuffer = []
+
+
+class ScriptOutputThread(Thread):
+
+    def __init__(self, server_address: Tuple[str, int], output: Queue):
+        super().__init__()
+        self.server_address = server_address
+        self.server = LogServer(server_address, output)
+        self.finished = False
+
+    def run(self):
+        try:
+            while True:
+                asyncore.loop(timeout=1, count=1)
+        finally:
+            self.server.close()
+            del self.server
+            asyncore.close_all()
+
+
+def output_service(queue: Queue, server: Optional[str], port: Optional[int]):
+    """Start a standalone output service
+
+    This service can be used in an other Python or R instance, for
+    Python instances the connection parameter externalClient need to
+    be specified.
+    """
+    try:
+        host = socket.gethostbyname(socket.gethostname())
+    except:
```

I would prefer "except Except" than the bare except